src/main/scala/ScalaModulePlugin.scala

```diff
@@ -147,6 +147,17 @@ object ScalaModulePlugin extends AutoPlugin {
 
   lazy val scalaModuleSettingsJVM: Seq[Setting[_]] = scalaModuleOsgiSettings
 
+  lazy val scalaModuleSettingsNative: Seq[Settings[_]] = Seq(
+    scalaVersion := "2.11.11",
+    skip in compile := System.getProperty("java.version").startsWith("1.6"),
+    test := {},
+    libraryDependencies := {
+      if (!scalaVersion.value.startsWith("2.11"))
+        libraryDependencies.value.filterNot(_.organization == "org.scala-native")
+      else libraryDependencies.value
+    }
```

Comment:
That does not make any sense. People with Scala.js cross-projects can publish the JVM part of their libraries for new binary-incompatible versions of Scala before Scala.js has been released for those new versions. And that's exactly the same resolve mechanism used by Scala Native (I should know, I co-designed `sbt-crossproject` as a generalization of Scala.js' cross-project).

You must simply *not run `fooNative/compile`* at all when you have done `++2.12.2`. It's as simple as that.

The most likely reason you think this doesn't work is because you `aggregate` all the `fooJVM`/`fooJS`/`fooNative` in a single `foo`, and then of course `foo/test` will insist on running `fooNative/test`. The solution is simple: do not use aggregation in those contexts.